pandas/core/series.py

```diff
@@ -4332,6 +4335,69 @@ def between(self, left, right, inclusive=True) -> "Series":
 
         return lmask & rmask
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_dtype(self: ABCSeries, convert_integer: bool = True) -> ABCSeries:
+        """
+        Convert columns of DataFrame or a Series to types supporting ``pd.NA``.
+
+        If the dtype is ``object``, if possible, convert to ``StringDtype``,
+        ``BooleanDtype`` or an appropriate integer extension type, otherwise leave as
+        ``object``. 
+
+        If the dtype is "integer", convert to an appropriate integer type.
+
+        If the dtype is numeric, and consists of all integers, convert to an
+        appropriate integer type.
+
+        Parameters
+        ----------
+        convert_integer : bool, default True
+            Whether ``int`` types should be converted to integer extension types
+
+        Returns
+        -------
+        converted : same type as input object
+        """
+        target_int_dtype = "Int64"
+
+        try:
+            inferred_dtype = lib.infer_dtype(self)
+        except ValueError:
```

Comment:
when does infer_dtype return a ValueError? (I don't you had this try/except in the previous iteration?)

Reply:
For `Period`.  See #23553
```python
In [1]: import pandas as pd

In [2]: pd.api.types.infer_dtype(pd.period_range("1/1/2011", freq="M", periods=3))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
c:\Code\pandas_dev\pandas\pandas\_libs\lib.pyx in pandas._libs.lib.infer_dtype()
   1243         try:
-> 1244             values = getattr(value, '_values', getattr(value, 'values', value))
   1245         except TypeError:

TypeError: Cannot convert PeriodArray to numpy.ndarray

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-2-df679b989f28> in <module>
----> 1 pd.api.types.infer_dtype(pd.period_range("1/1/2011", freq="M", periods=3))

c:\Code\pandas_dev\pandas\pandas\_libs\lib.pyx in pandas._libs.lib.infer_dtype()
   1251
   1252             # its ndarray like but we can't handle
-> 1253             raise ValueError(f"cannot infer type for {type(value)}")
   1254
   1255     else:

ValueError: cannot infer type for <class 'NoneType'>
```

Reply:
That looks like https://github.com/pandas-dev/pandas/issues/23553